### DISCUSSION TITLE
Not getting current access state of location pemission iOS

### DISCUSSION THREAD

**Discussion by Dipannita20** [shinyorg/shiny#906] [Q&A] [answered]
iOS 15.3
If user deny **Location permission to deny** in first place, 2nd time below code is infinite wait.

`                 var accessState = await GeofenceManager.RequestAccess();
`
Same happens with **Allow Once**
Is it possible to get current access state instead of infinite waiting?

Thanks in advance.

**Comment by aritchie** [comment, marked as answer]
RequestAccess is a one stop for all things relating to the particular functionality being available.  If you have a bug, please file an issue.  Please do not use the forum to skip that process!

As it is, I discovered this the other day as well and there is a fix on preview 2.4.3 already

**Reply by Dipannita20** [reply]
Thanks for the response, upgrading to 2.4.3 resolve the issue. 
But after upgrading getting into another interesting senario, **on iOS 15.3 and shiny 2.4.3 not showing 2nd location permission popup for  **Always****, I can only provide permission  **"While Using the App"**.  Is it an issue or expected behaviour?

`ShinyHost.Resolve<IGeofenceManager>().RequestAccess();
`
I am happy to raise a bug if I am sure about it. 
Thanks in advanced

**Reply by aritchie** [reply]
You are becoming a support burden to this project.  Please remember that Shiny is NOT a product.  You need to spend some time reading through the source code, checking the samples, reading the platform documentation for the OS.  The iOS documentation covers this.  The answer is "yes - this is expected".

This is not the place for issues because it never contains enough info.  So I'm afraid it was not an ask - file the bug if you have one with proper details.

The flow of benefit has been completely one directional between the support, use of the libraries professionally, etc.  Please start contributing back to this library in some form (bugfixes, sponsorship, etc).

**Reply by Dipannita20** [reply]
Thanks, looking forward to contributing to library.
